be1-go/message/test/messagedata/authenticate_user_test.go

```diff
@@ -0,0 +1,60 @@
+package messagedata
+
+import (
+	"encoding/json"
+	"github.com/stretchr/testify/require"
+	"os"
+	"path/filepath"
+	"popstellar/message/messagedata"
+	"testing"
+)
+
+// TestAuthUserInterfaceFunctions tests the basic interface methods of messagedata
+func TestAuthUserInterfaceFunctions(t *testing.T) {
+	var authMsg messagedata.AuthenticateUser
+	require.Equal(t, messagedata.AuthObject, authMsg.GetObject())
+	require.Equal(t, messagedata.AuthAction, authMsg.GetAction())
+	require.Empty(t, authMsg.NewEmpty())
+}
+
+// TestVerify runs multiple times the verification method on different jsons
+func TestVerify(t *testing.T) {
+
+	var authUser messagedata.AuthenticateUser
+
+	// action and object are constant
+	object, action := "popcha", "authenticate"
+
+	// generic method, takes a boolean argument, to assert whether the given json should be valid or not
+	getTestValid := func(file string, shouldBeValid bool) func(*testing.T) {
+		return func(t *testing.T) {
+			// read the bad example file
```

Comment:
This function works with both good and bad examples, this comment is therefore wrong

Reply:
Indeed, i will change that!